### PR TITLE
Navbar and Bookapp

### LINKED ISSUE #15: First Implementation of Navbar + Full Psychiatrist Profile
Inside the components folder, create the following components:

a. `Navbar.tsx`: Based on current mid-fis, this component should show a login button (directs to signInWithPopUp) and an about tab, that should direct to the `about-us` page. Create this page if it doesn't already exist in `pages`.

b. `BookApp.tsx`: Adhere to the current mid-fis, and the picture below. You should not have to implement the dropdown calendar for now, but test the toggle functionality of each button.

For reference:

Image

- [x] MOVE TICKET ON BOARD
- [x] `Navbar.tsx`
- [x] `BookApp.tsx`
- [x] This ticket is pretty light, compared to others. Read up on WebRTC, peerJS, and Websockets. More structured details coming soon, but expect to **make a presentation** with findings to our entire subteam next meeting.

### PR DESCRIPTION
**Pull request by hahnandrew**
## Overview

- Added navbar component (Navbar.tsx)
- Added bookapp component (BookApp.tsx)

resolves #15

## Changes Made

```
function Navbar
  return (
              <Link href="/About-Us" className="">About Us</Link>
              <button onClick={signInWithGoogle}>Login</button>
 )
```
```
function BookApp
  return (
            Book Appointment Now
          <Dropdown img={<CalendarIcon />} text="Appointment Date" select="Select Day" />
          <Dropdown img={<ClockIcon />} text="Appointment Time" select="Select Time" />
          <Dropdown img={<CameraIcon />} text="Video Call" select="Select Platform" />
          <TempCalendar />
          <button><p>Next →</p></button>
 )
```

## Test Coverage

ran navbar and bookapp components on index.tsx

testing

login

finalized

## Next Steps 

We will convert this to high fidelity when designs are finalized

## Replication
```
cd frontend
yarn install
yarn run dev
```

### CODE REVIEW

File: frontend/src/pages/index.tsx
Comment (biojameskim) [inline frontend/src/pages/index.tsx:1]: Since this file was just used for testing, it might be good to not push this.

File: frontend/src/styles/globals.css
Comment (biojameskim) [inline frontend/src/styles/globals.css:1]: I'm not sure if we'll be using the other css styling code other than the tailwind imports. Maybe we can delete these?

File: frontend/src/Components/BookApp.tsx
Comment (biojameskim) [inline frontend/src/Components/BookApp.tsx:1]: I'm assuming you guys didn't style things too detailed because we aren't sure what the final designs will look like. In that case, looks good to me!

File: frontend/dist/output.css
Comment (biojameskim) [inline frontend/dist/output.css:1]: Where is this file being used? If it's not being used maybe we can delete this.

File: frontend/next.config.js
Comment (biojameskim) [inline frontend/next.config.js:1]: What is the webpack config exactly for? I think some documentation here would be helpful.

File: frontend/contexts/AuthContext.tsx
Comment (biojameskim) [inline frontend/contexts/AuthContext.tsx:1]: The helpful error message looks great!

**Review by biojameskim — CHANGES_REQUESTED** [review]
Overall, looks great to me! I left some comments on certain files regarding documentation and unused code.

File: frontend/src/styles/globals.css
Comment (sriramanved) [reply frontend/src/styles/globals.css:1]: Yes

File: frontend/next.config.js
Comment (sriramanved) [reply frontend/next.config.js:1]: What is this file used for?

File: frontend/next.config.js [hunk 1]
```diff
@@ -3,6 +3,15 @@ const { default: next } = require('next')
 /** @type {import('next').NextConfig} */
 const nextConfig = {
   reactStrictMode: true,
+  webpack(config) {
+    config.module.rules.push({
+      test: /\.svg$/i,
+      issuer: /\.[jt]sx?$/,
+      use: ['@svgr/webpack'],
```
Comment (sriramanved) [inline frontend/next.config.js:11]: Do you need this?

File: frontend/src/pages/index.tsx [hunk 2]
```diff
@@ -14,110 +15,11 @@ export default function Home() {
...
-            </h2>
-            <p className={inter.className}>
-              Instantly deploy your Next.js site to a shareable URL
-              with&nbsp;Vercel.
-            </p>
-          </a>
-        </div>
+      <main>
+        <Navbar />
+        <BookApp />
```
Comment (sriramanved) [inline frontend/src/pages/index.tsx]: This should not be in your PR. This should only be for testing your feature.

File: frontend/src/styles/globals.css [hunk 3]
```diff
@@ -9,30 +13,24 @@
...
-    rgba(255, 255, 255, 1),
-    rgba(255, 255, 255, 0)
-  );
+  --primary-glow: conic-gradient(from 180deg at 50% 50%,
+      #16abff33 0deg,
+      #0885ff33 55deg,
+      #54d6ff33 120deg,
+      #0071ff33 160deg,
+      transparent 360deg);
+  --secondary-glow: radial-gradient(rgba(255, 255, 255, 1),
```
Comment (sriramanved) [inline frontend/src/styles/globals.css]: Why did you add these?

File: frontend/src/Components/BookApp.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+          <p className="font-bold">
+            Book Appointment Now
+          </p>
+        </div>
+        <div className="flex flex-row">
+          {/* <div className="shadow-lg"></div> */}
+          <Dropdown img={<CalendarIcon />} text="Appointment Date" select="Select Day" />
+          <Dropdown img={<ClockIcon />} text="Appointment Time" select="Select Time" />
+          <Dropdown img={<CameraIcon />} text="Video Call" select="Select Platform" />
+        </div>
```
Comment (sriramanved) [inline frontend/src/Components/BookApp.tsx]: It's fine if you have used pre-built components, but the styling of the components is still not good. You need to fix this and make sure it is actually similar to the Mid-fis.

**Review by sriramanved — CHANGES_REQUESTED** [review]
Good start, please see the attached comments and iterate on this.

**Comment by DylanTom** [comment]
image

File: frontend/src/styles/globals.css (hunk 3 above)
Comment (DylanTom) [reply frontend/src/styles/globals.css]: Resolved w/ afd787b

File: frontend/next.config.js (hunk 1 above)
Comment (DylanTom) [reply frontend/next.config.js:11]: Yes, documentation added in afd787b

File: frontend/next.config.js
Comment (DylanTom) [reply frontend/next.config.js:1]: Allows us to add SVG to our project assets

File: frontend/dist/output.css
Comment (DylanTom) [reply frontend/dist/output.css:1]: Deleted in afd787b

File: frontend/src/Components/BookApp.tsx
Comment (DylanTom) [reply frontend/src/Components/BookApp.tsx:1]: Working on styling in afd787b

File: frontend/src/styles/globals.css
Comment (DylanTom) [reply frontend/src/styles/globals.css:1]: Resolved in afd787b

File: frontend/src/pages/index.tsx
Comment (DylanTom) [reply frontend/src/pages/index.tsx:1]: Resolved with afd787b

**Comment by DylanTom** [comment]
image

File: frontend/src/Components/BookApp.tsx (hunk 4 above)
Comment (DylanTom) [reply frontend/src/Components/BookApp.tsx]: Resolved 780d426

File: frontend/src/pages/index.tsx (hunk 2 above)
Comment (DylanTom) [reply frontend/src/pages/index.tsx]: Resolved 780d426

**Comment by sriramanved** [comment]
Good work! Please note tips we talked about in subteam.

**Review by sriramanved — APPROVED** [review]
Changes approved!
